frame/treasury/src/lib.rs

```diff
@@ -168,7 +168,7 @@ decl_module! {
 		/// # </weight>
 		#[weight = SimpleDispatchInfo::FixedOperational(100_000)]
 		fn reject_proposal(origin, #[compact] proposal_id: ProposalIndex) {
-			T::RejectOrigin::ensure_origin(origin).map_err(|e| Into::<&str>::into(e))?;
+			T::RejectOrigin::ensure_origin(origin).map_err(|e| e.as_str())?;
```

Comment:
why does it need to be `.as_str()`? can't the error be returned back directly?

Reply:
I think `ensure_origin` returns `system::Error` which needs to be converted to `treasury::Error` accordingly to work but I think `Into::<&str>::into(e)` should still work.